### PR TITLE
use last event

### PR DESCRIPTION
**Pull request by atomraymaker**
**Issue**
This function updates the orchestrator status table, by bubbling up status to the step, activity, and workflow. DynamoDB is providing batches of updates to process. In those batches, there are multiple updates that deal with the same record, which is causing a lot of writes to the same record in a small period of time.

**Solution**
Use the last DDB event for each UID. The event includes a full copy of the latest state so it should work to base any updates on. This means that all interim updates are ignored and collapsed into responding to this one event.

### CODE REVIEW

File: __mock__/mockDynamoDb.js [hunk 1]
```diff
@@ -13,83 +24,90 @@ class MockDynamoDb {
...
-    }
-    get(params) {
+        this.updateInputs = [];
+    };
+    MockDynamoDb.prototype.get = function (params) {
+        var _this = this;
         return {
-            promise: () => {
-                return new Promise((resolve, reject) => {
+            promise: function () {
```
Comment (nkeating-mutualofenumclaw) [inline __mock__/mockDynamoDb.js]: This conversion seems like its doing the same thing with a slightly different syntax.  Is there a specific reason why this change should be made

File: __mock__/mockDynamoDb.js [hunk 2]
```diff
@@ -13,83 +24,90 @@ class MockDynamoDb {
...
+                return new Promise(function (resolve, reject) {
                     resolve({
                         '$response': {
-                            error: this.error,
+                            error: _this.error
```
Comment (nkeating-mutualofenumclaw) [inline __mock__/mockDynamoDb.js]: Typescript resolves this, where javascript wouldn't.  There's no need for the _this resolution

File: __mock__/mockDynamoDb.js (hunk 1 above)
Comment (atomraymaker) [reply __mock__/mockDynamoDb.js]: this was autogenerated from a change to the .ts file

Comment (nkeating-mutualofenumclaw) [reply __mock__/mockDynamoDb.js]: ah, okay

**Comment by nkeating-mutualofenumclaw** [comment]
Looks good
